code/game/jobs/job_objective.dm

```diff
@@ -1,69 +1,71 @@
-/datum/mind/var/list/job_objectives = list()
-
-#define FINDJOBTASK_DEFAULT_NEW 1 // Make a new task of this type if one can't be found.
-/datum/mind/proc/findJobTask(typepath, options = 0)
-	var/datum/job_objective/task = locate(typepath) in job_objectives
-	if(!istype(task,typepath))
-		if(options & FINDJOBTASK_DEFAULT_NEW)
-			task = new typepath()
-			job_objectives += task
-	return task
+/datum/mind/proc/find_job_task(typepath)
+	for(var/datum/job_objective/task as anything in job_objectives)
```

Comment:
A `locate(typepath) in job_objectives` should work here

Reply:
done